File: docs/examples/py/src/buckets_create.py
from reduct import Client, BucketSettings, QuotaType




async def create_bucket():
    # Create a client with the base URL and API token
    with Client("http://localhost:8383", api_token="my-token") as client:
        # Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
        settings = BucketSettings(
            quota_type=QuotaType.FIFO,
            quota_size=1000_000_000,
        )
        bucket = await client.create_bucket("my-bucket", settings, exist_ok=True)
        assert bucket.name == "my-bucket"


if __name__ == "__main__":
    import asyncio

    asyncio.run(create_bucket())

File: docs/examples/go/src/buckets_create.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Create a client using the server URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
	settings := model.NewBucketSettingBuilder().
		WithQuotaType(model.QuotaTypeFifo).
		WithQuotaSize(1_000_000_000).
		Build()
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", &settings)
	if err != nil {
		panic(err)
	}

	if bucket.Name != "my-bucket" {
		panic("Bucket was not created or retrieved correctly")
	}
}
